src/rgw/rgw_pubsub.cc

```diff
@@ -512,8 +587,29 @@ int RGWPubSub::Bucket::write_topics(const DoutPrefixProvider *dpp, const rgw_pub
   return 0;
 }
 
-int RGWPubSub::get_topic(const DoutPrefixProvider *dpp, const std::string& name, rgw_pubsub_topic& result, optional_yield y) const
-{
+int RGWPubSub::get_topic(const DoutPrefixProvider* dpp,
+                         const std::string& name,
+                         rgw_pubsub_topic& result,
+                         optional_yield y,
+                         bool fetch_bucket_mapping) const {
+  if (driver->get_zone()->get_zonegroup().supports_feature(
+          rgw::zone_features::notification_v2)) {
+    int ret = driver->read_topic_v2(name, tenant, result, nullptr, y, dpp);
+    if (ret < 0) {
+      ldpp_dout(dpp, 1) << "failed to read topic info for name: " << name
+                        << " tenant: " << tenant << ", ret=" << ret << dendl;
```

Comment:
```suggestion
                        << " tenant: " << tenant << ", ret=" << ret << dendl;
    return ret;                        
```
if the topic does not exist we should not proceed. note that without that fix, the `test_ps_s3_notification_errors_on_master` test is failing

Reply:
> if the topic does not exist we should not proceed. note that without that fix, the `test_ps_s3_notification_errors_on_master` test is failing

thanks, done.